src/contrib/mlir/runtime/cpu/cpu_runtime.cpp

```diff
@@ -142,33 +157,57 @@ void MLIRCPURuntime::cleanup()
     // Free void double pointer arguments without freeing external tensor data.
     for (auto* arg : m_invokeArgs)
     {
-        auto* memRefArg = *(reinterpret_cast<StaticMemRef**>(arg));
-        free(memRefArg);
-        free(arg);
+        if (!clEnableBarePtrMemRefLowering)
+        {
+            // Default memref lowering lowers memrefs to StaticMemRef descriptors.
+            auto* memRefArg = *(reinterpret_cast<StaticMemRef**>(arg));
+            free(memRefArg);
+            free(arg);
+        }
+        else
+        {
+            // Custom memref lowering lowers memref arguments to bare pointers to tensors.
+            auto** memRefArg = reinterpret_cast<void**>(arg);
+            free(memRefArg);
+        }
     }
 }
 
-// The current call ABI takes a single arg pointer (argPtr) pointing to a list of args.
+// The default call ABI takes a single arg pointer (argPtr) pointing to a list of args.
 // Each arg is a  pointer to a StaticMemRef which contains a data pointer
 //
 // The args are laid out as follows
 // argPtr-> arg[0]-> StaticMemRef -> <data>
 //          arg[1]-> StaticMemRef -> <data>
 //          ...
+//
+// The bare pointer ABI takes a single arg pointer pointing to data for that MemRef. Not extra
+// information about the MemRef is passed at the moment.
```

Comment:
nit: can you provide an example countering the one above ?

Reply:
Is the example above still valid? Don't we pass `arg0`, `arg1`, etc., directly, instead of through `argPtr`